cmd/domain.go

```diff
@@ -57,11 +59,26 @@ func domainScannerLogic(ctx *cobra.Command, readr io.ReadWriteSeeker) []rdap.Who
 	rd, _ := time.ParseDuration(r)
 	utils.SizeCheck(readr)
 	scanner := bufio.NewScanner(readr)
+ScannerLoop:
 	for scanner.Scan() {
 		if i := scanner.Text(); i != "" {
 			cli.Info("Searching Whois Records for Domain %v", i)
-			if r, e := queryDomain(i); e != nil {
-				cli.Errorln("Whois lookup error", e)
+			// Get root domain
+			rootDomain, err := getRootDomain(i)
+			if err != nil {
+				cli.Errorln("Root Domain Error: ", err)
+			}
+			// Check if root romain has been queried
+			for index, entry := range result {
+				if entry.Name == strings.ToUpper(rootDomain) {
+					//Root domain has already been queried, add to root domain entry "DomainsSearched" .
+					result[index].DomainsSearched = append(result[index].DomainsSearched, strings.ToUpper(i))
+					//Skip query and exit out of current iteration of ScannerLoop.
+					continue ScannerLoop
+				}
+			}
+			if r, e := queryDomain(rootDomain); e != nil {
```

Comment:
The first requested domain which is converted to a rootDomain is not added to the result struct.

Reply:
Updated with additional commit to resolve issue